merkle/common.go

```diff
@@ -32,3 +32,37 @@ func bit(index []byte, i int) uint {
 	bIndex := (IndexBits - i - 1) / 8
 	return uint((index[bIndex] >> uint(i%8)) & 0x01)
 }
+
+// flipBit returns index with the i'th bit from the right flipped.
+func flipBit(index []byte, i int) []byte {
+	r := make([]byte, len(index))
+	copy(r, index)
+	IndexBits := len(index) * 8
+	bIndex := (IndexBits - i - 1) / 8
+	r[bIndex] ^= 1 << uint(i%8)
+	return r
+}
+
+// Neighbor returns index with only the left i bits set and the i'th bit flipped.
```

Comment:
Can you give a bit more details on this? In particular we have a Sibling() function for logs, which could lead to confusion.